### ISSUE TITLE
TPOT OneHotEncoder cannot fit more than once to different datasets

### ISSUE THREAD

**Issue by JameyPMcDowell** [EpistasisLab/tpot#1050]
TPOT OneHotEncoder cannot refit on a new dataset

## Context of the issue

Given X1 and X2, TPOT cannot call fit(X1), then fit(X2).  This breaks down the ability to refit on full datasets after using an exported_pipeline to perform cross-validation, without having to clone the estimator

#1049 # Process to reproduce the issue

>>> from tpot.builtins import OneHotEncoder
>>> import numpy as np

>>> X1 = np.array(
>>>    [[1, 0, 0, 0],
>>>     [0, 1, 0, 0],
>>>     [0, 0, 1, 0],
>>>     [0, 0, 0, 1]]
>>> )

>>> X2 = np.array(
>>>     [[1, 0, 0, 0, 0],
>>>      [0, 1, 0, 0, 0],
>>>      [0, 0, 1, 0, 0],
>>>      [0, 0, 0, 1, 0],
>>>      [0, 0, 0, 0, 1]]
>>> )

>>> ohe = OneHotEncoder()

>>> ohe.fit(X1)
OneHotEncoder(categorical_features=[True, True, True, True],
              dtype=<class 'float'>, minimum_fraction=None, sparse=True,
              threshold=10)

>>> ohe.fit(X2)
IndexError: boolean index did not match indexed array along dimension 0; dimension is 5 but corresponding boolean dimension is 4
 

## Expected result

>>> ohe.fit(X1)
OneHotEncoder(categorical_features=[True, True, True, True],
              dtype=<class 'float'>, minimum_fraction=None, sparse=True,
              threshold=10)

>>> ohe.fit(X2)
OneHotEncoder(categorical_features=[True, True, True, True, True],
              dtype=<class 'float'>, minimum_fraction=None, sparse=True,
              threshold=10)

## Current result
>>> ohe.fit(X2)

IndexError: boolean index did not match indexed array along dimension 0; dimension is 5 but corresponding boolean dimension is 4

## Possible fix

In fit_transform(), if self.categorical_features == 'auto', assign the output of auto_select_categorical_features to a local variable, and call _transform_selected(selected=local_variable)

**Comment by weixuanfu** [comment]
Thank you for reporting the bug and suggestion. I just submitted a PR for this issue and will merge it the dev branch.

**Comment by JameyPMcDowell** [comment]
Can we go ahead and merge just this fix into master? It's breaking our codebase when we go from scoring in cross-validation to fitting on our full dataset

**Comment by weixuanfu** [comment]
Hmm, why not just using dev branch? Does dev branch will also break your codebase?

Just FYI, you may try to install the dev branch to ur environment via the command below:
```
pip install --upgrade --no-deps --force-reinstall git+https://github.com/EpistasisLab/tpot.git@development
```

We eventually will merge dev branch to master branch after more testing as a minor release.

**Comment by JameyPMcDowell** [comment]
thanks for the quick install upgrade.  was worried i might bring a bug from dev into our production system, but if coverage is good in dev, then we'll go with that until next minor release.  Thanks for your response times on this, really appreciate TPOT and the work y'all do on it

**Comment by weixuanfu** [comment]
This issue should be fixed in TPOT v0.11.2. Please feel free to reopen this issue if there is a unsolved issue related to this one.
